File: src/main/java/com/pengrad/telegrambot/model/Voice.java
package com.pengrad.telegrambot.model;

/**
 * stas
 * 10/21/15.
 */
public class Voice {

    private String file_id;
    private Integer duration;
    private String mime_type;
    private Integer file_size;

    Voice() {
    }

    public String fileId() {
        return file_id;
    }

    public Integer duration() {
        return duration;
    }

    public String mimeType() {
        return mime_type;
    }

    public Integer fileSize() {
        return file_size;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Voice voice = (Voice) o;

        if (file_id != null ? !file_id.equals(voice.file_id) : voice.file_id != null) return false;
        if (duration != null ? !duration.equals(voice.duration) : voice.duration != null) return false;
        if (mime_type != null ? !mime_type.equals(voice.mime_type) : voice.mime_type != null) return false;
        return file_size != null ? file_size.equals(voice.file_size) : voice.file_size == null;
    }

    @Override
    public int hashCode() {
        return file_id.hashCode();
    }

    @Override
    public String toString() {
        return "Voice{" +
                "file_id='" + file_id + '\'' +
                ", duration=" + duration +
                ", mime_type='" + mime_type + '\'' +
                ", file_size=" + file_size +
                '}';
    }
}

File: core/src/main/kotlin/com/pengrad/telegrambot/model/Voice.kt
package com.pengrad.telegrambot.model

import java.io.Serializable

/**
 * stas
 * 10/21/15.
 */
class Voice : Serializable {

    private val file_id: String? = null
    private val duration: Int? = null
    private val mime_type: String? = null
    private val file_size: Int? = null

    fun fileId(): String? {
        return file_id
    }

    fun duration(): Int? {
        return duration
    }

    fun mimeType(): String? {
        return mime_type
    }

    fun fileSize(): Int? {
        return file_size
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val voice = o as Voice?

        if (if (file_id != null) file_id != voice!!.file_id else voice!!.file_id != null) return false
        if (if (duration != null) duration != voice.duration else voice.duration != null) return false
        if (if (mime_type != null) mime_type != voice.mime_type else voice.mime_type != null) return false
        return if (file_size != null) file_size == voice.file_size else voice.file_size == null
    }

    override fun hashCode(): Int {
        return file_id?.hashCode() ?: 0
    }

    override fun toString(): String {
        return "Voice{" +
                "file_id='" + file_id + '\''.toString() +
                ", duration=" + duration +
                ", mime_type='" + mime_type + '\''.toString() +
                ", file_size=" + file_size +
                '}'.toString()
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}